extensions/cms/src/cmsUtils.ts

```diff
@@ -122,7 +173,20 @@ export class CmsUtils {
 			connection: connection,
 			ownerUri: ownerUri
 		};
+
+		// update a server if a server with same name exists
+		this._registeredCmsServers = this._registeredCmsServers.filter((server) => {
+			return server.name !== name;
+		});
 		this._registeredCmsServers.push(cmsServerNode);
+
+		// save the CMS Servers for future use
+		let toSaveCmsServers: ICmsResourceNodeInfo[] = JSON.parse(JSON.stringify(this._registeredCmsServers));
+		toSaveCmsServers.forEach(server => {
+			server.ownerUri = undefined;
+			server.connection.options.password = '';
+		});
```

Comment:
Explain why you're clearing the password here

Reply:
This is the same logic followed before. We don't want to save the passwords in config.

Reply:
I'm saying put that in a comment so other people don't wonder why this is being done.

Reply:
This still could be made more useful...if I wasn't familiar with the code my first question would be "Why aren't we? What config? Why does it matter if the password is stored in this config?"

Something like this is much more useful...

// Don't save the password since this is persisted as a plain text file to disk

---
In reply to: [292671856](https://github.com/microsoft/azuredatastudio/pull/5989#discussion_r292671856) [](ancestors = 292671856)